src/snowflake/snowpark/mock/_plan.py

```diff
@@ -561,8 +561,8 @@ def execute_mock_plan(
         for column_name in table.columns:
             sf_type = table.sf_types[column_name]
             table[column_name].sf_type = table.sf_types[column_name]
-            if not isinstance(sf_type.datatype, _NumericType):
-                table[column_name].replace(np.nan, None, inplace=True)
+            if isinstance(sf_type.datatype, _NumericType):
```

Comment:
looks like you remove the `not` here

Reply:
@sfc-gh-aalam : yes I did.  The old version had no effect for multiple reasons.  I was trying to determine what the original design intent was.  np.nan is a float type, so this would only make sense if it is applied to numeric types. One could also just fill Nan on the whole dataset for that matter.  

I am guessing that the idea here is to model the data which would be returned by a real snowflake connection.  I am pretty sure you would not see numpy anywhere there because numpy is an optional dependency I believe.

Reply:
`NaN` is a valid value for Snowflake Float type, so this line should be `if not isinstance(sf_type.datatype, (DoubleType, FloatType)):`. 
We replace `np.nan` with `None` for other types because Pandas converts `None` to `nan` automatically if the dtype is something like `np.int64`, but for Snowflake integer columns should only contain `null` (mapped to `None` in Python) instead of `nan`.